airbyte-integrations/connectors/source-hubspot/source_hubspot/components.py

```diff
@@ -0,0 +1,48 @@
+#
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+#
+
+from typing import Any, Dict, Optional
+
+from airbyte_cdk.sources.declarative.transformations import RecordTransformation
+from airbyte_cdk.sources.types import Config, StreamSlice, StreamState
+
+
+class NewtoLegacyFieldTransformation(RecordTransformation):
+    """
+    Implements a custom transformation which adds the legacy field equivalent of v2 fields for streams which contain Deals and Contacts entities.
+
+    This custom implmentation was developed in lieu of the AddFields component due to the dynamic-nature of the record properties for the HubSpot source. Each
+
+    For example:
+    hs_v2_date_exited_{stage_id} -> hs_date_exited_{stage_id} where {stage_id} is a user-generated value
+    """
+
+    def __init__(self, field_mapping: Dict[str, str]) -> None:
+        self._field_mapping = field_mapping
+
+    def transform(
+        self,
+        record_or_schema: Dict[str, Any],
+        config: Optional[Config] = None,
+        stream_state: Optional[StreamState] = None,
+        stream_slice: Optional[StreamSlice] = None,
+    ) -> None:
+        """
+        Transform a record in place by adding fields directly to the record by manipulating the injected fields into a legacy field to avoid breaking syncs.
```

Comment:
nit: the comment should mention its both records or the catalog schema

Reply:
Agreed. Updated.